server/filesystem/filesystem.go

```diff
@@ -329,17 +329,8 @@ func (fs *Filesystem) Copy(p string) error {
 		return err
 	}
 
-	base := info.Name()
-	extension := filepath.Ext(base)
-	baseName := strings.TrimSuffix(base, extension)
-
-	// Ensure that ".tar" is also counted as apart of the file extension.
-	// There might be a better way to handle this for other double file extensions,
-	// but this is a good workaround for now.
-	if strings.HasSuffix(baseName, ".tar") {
-		extension = ".tar" + extension
-		baseName = strings.TrimSuffix(baseName, ".tar")
-	}
+	baseName := info.Name()
```

Comment:
The issue with building is caused here. You removed the base and are passing `baseName` which still has the extension. The original code trims that from the name before passing it to the find copy suffix.

This is what I have for 332-334 now and the tests pass.

```go
	base := info.Name()
	extension := fs.Ext(base)
	baseName := strings.TrimSuffix(base, extension)
```